apps/authentications/src/app/app.service.ts

```diff
@@ -596,4 +607,22 @@ export class AppService {
       );
     }
   }
+
+  /**
+   * Get country and continentalCode from http://ip-api.com/json/
+   * @param {string} ip
+   * @returns {CheckIpDto}
+   */
+  async getGeolocationFromIp(ip: string) {
+    try {
+      const result = await lastValueFrom(
+        this.httpService
+          .get<CheckIpDto>(getIPUrl(ip))
+          .pipe(map(({ data }) => data))
+      );
+      return result;
```

Comment:
```suggestion
      return lastValueFrom(
        this.httpService
          .get<CheckIpDto>(getIPUrl(ip))
          .pipe(map(({ data }) => data))
      );
```
ref. https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/docs/rules/prefer-immediate-return.md